core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -560,16 +587,30 @@ private[kafka] class Processor(val id: Int,
   /**
    * Queue up a new connection for reading
    */
-  def accept(socketChannel: SocketChannel) {
-    newConnections.add(socketChannel)
-    wakeup()
+  def accept(socketChannel: SocketChannel,
+             mayBlock: Boolean): Boolean = {
+    val accepted = {
+      if (newConnections.offer(socketChannel)) {
+        true
+      } else if (mayBlock) {
+        val startNs = time.nanoseconds
```

Comment:
I don't think `startNS` is use here? Previously it was used to mark the `acceptorIdlePercentMeter`